src/Symfony/Component/Serializer/NameConverter/MetadataAwareNameConverter.php

```diff
@@ -114,8 +119,12 @@ private function getCacheValueForAttributesMetadata(string $class, array $contex
         $classMetadata = $this->metadataFactory->getMetadataFor($class);
 
         $cache = [];
+        $groups = [];
+        if (isset($context['groups'])) {
+            $groups = \is_array($context['groups']) ? $context['groups'] : [$context['groups']];
```

Comment:
```suggestion
            $groups = (array) $context['groups']);
```

Reply:
I changed a little the code to use casting but need to do an isset to be sure groups key is here, otherwise it throws a warning
done